src/article/services/article.service.ts

```diff
@@ -57,6 +57,57 @@ export class ArticleService {
     });
   }
 
+  async getArticles(
+    ctx: RequestContext,
+    limit: number,
+    offset: number,
+  ): Promise<{ articles: ArticleOutput[]; count: number }> {
+    this.logger.log(ctx, `${this.getArticles.name} was called`);
+
+    const actor: Actor = ctx.user;
+
+    const isAllowed = this.aclService.forActor(actor).canDoAction(Action.List);
+    if (!isAllowed) {
+      throw new UnauthorizedException();
+    }
+
+    this.logger.log(ctx, `calling ${ArticleRepository.name}.findAndCount`);
+    const [articles, count] = await this.repository.findAndCount({
+      where: {},
+      take: limit,
+      skip: offset,
+    });
+
+    const articlesOutput = plainToClass(ArticleOutput, articles, {
+      excludeExtraneousValues: true,
+    });
+
+    return { articles: articlesOutput, count };
+  }
+
+  async getArticleById(
+    ctx: RequestContext,
+    id: number,
+  ): Promise<ArticleOutput> {
+    this.logger.log(ctx, `${this.getArticleById.name} was called`);
+
+    const actor: Actor = ctx.user;
+
+    this.logger.log(ctx, `calling ${ArticleRepository.name}.getById`);
+    const article = await this.repository.getById(id);
+
+    const isAllowed = this.aclService
+      .forActor(actor)
+      .canDoAction(Action.Read, article);
```

Comment:
```suggestion
      .canDoAction(Action.Read);
```
Check which is better. 
Since the article will not actually be used. 
The `Action.Read` does not have any callback function.